inspirehep/modules/migrator/legacy/audit.py

```diff
@@ -24,3 +24,38 @@
 """INSPIRE migrator load/dump audit."""
 
 from __future__ import absolute_import, print_function
+
+import json
+
+from sqlalchemy.orm.exc import NoResultFound#, MultipleResultsFound
+
+def date_handler(obj):
+    return obj.isoformat() if hasattr(obj, 'isoformat') else obj
+
+def get(*args, **kwargs):
+    """Get audits."""
+    from inspirehep.modules.audit.models import Audit
+    q = Audit.query
+    return q.count(), q.all()
+
+
+def dump(u, from_date, **kwargs):
+    """Dump the audits as a list of dictionaries."""
+    from inspirehep.modules.audit.models import Audit
+
+    # First fetch all the audit information
+    audit = {}
+    try:
+        result = Audit.query.filter_by(id=u.id).one()
```

Comment:
Since you do on top `q.all()` you already retrieve all rows in the audit table. This function dump is then called with each item so you don't need to retrieve it again (or check if it is not present in the DB)